src/librbd/migration/NativeFormat.cc

```diff
@@ -53,36 +56,82 @@ std::string NativeFormat<I>::build_source_spec(
 template <typename I>
 NativeFormat<I>::NativeFormat(
     I* image_ctx, const json_spirit::mObject& json_object, bool import_only)
-  : m_image_ctx(image_ctx), m_json_object(json_object),
+  : m_image_ctx(image_ctx), m_cct(m_image_ctx->cct), m_json_object(json_object),
     m_import_only(import_only) {
 }
 
 template <typename I>
 void NativeFormat<I>::open(Context* on_finish) {
-  auto cct = m_image_ctx->cct;
-  ldout(cct, 10) << dendl;
+  ldout(m_cct, 10) << dendl;
+
+  std::optional<std::string> cluster_name;
+  auto& cluster_name_val = m_json_object[CLUSTER_NAME_KEY];
+  if (cluster_name_val.type() == json_spirit::str_type) {
+    cluster_name = cluster_name_val.get_str();
+  } else if (cluster_name_val.type() != json_spirit::null_type) {
+    lderr(m_cct) << "invalid cluster name" << dendl;
+    on_finish->complete(-EINVAL);
+    return;
+  }
+
+  std::optional<std::string> client_name;
+  auto& client_name_val = m_json_object[CLIENT_NAME_KEY];
+  if (client_name_val.type() == json_spirit::str_type) {
+    if (!cluster_name.has_value()) {
+      lderr(m_cct) << "client name takes effect only with cluster name"
+                   << dendl;
+      on_finish->complete(-EINVAL);
+      return;
+    }
+    client_name = client_name_val.get_str();
+  } else if (client_name_val.type() != json_spirit::null_type) {
+    lderr(m_cct) << "invalid client name" << dendl;
+    on_finish->complete(-EINVAL);
+    return;
+  } else {
+    if (cluster_name.has_value()) {
+      ldout(m_cct, 10) << "client name not set. assuming client.admin" << dendl;
+      client_name = "client.admin";
```

Comment:
It's better to avoid repeating defaults.  Let's make a small tweak to `init_rados()` and get rid of this `else`.